source/Calamari/Integration/Nginx/WindowsNginxServer.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Calamari.Integration.Nginx
+{
+    public class WindowsNginxServer : NginxServer
+    {
+        protected override string GetConfigRootDirectory()
+        {
+            throw new System.NotImplementedException();
```

Comment:
I tried running on a windows Tentacle and did not receive this exception.  Instead:

```
12:34:34   Error    |       System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\WINDOWS\system32\config\systemprofile\AppData\Local\Calamari\Temp\7f46408f-2d39-4cd7-b229-9e6b07b3440c\conf\dummy.conf'.
12:34:34   Error    |       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
12:34:34   Error    |       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
12:34:34   Error    |       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy)
12:34:34   Error    |       at System.IO.FileStream..ctor(String path, FileMode mode)
12:34:34   Error    |       at Calamari.Integration.FileSystem.CalamariPhysicalFileSystem.WriteAllText(String path, String contents)
12:34:34   Error    |       at Calamari.Integration.FileSystem.CalamariPhysicalFileSystem.RetryTrackerFileAction(Action fileAction, String target, String action)
12:34:34   Error    |       at Calamari.Integration.Nginx.NginxServer.SaveConfiguration(String tempDirectory)
12:34:34   Error    |       at Calamari.Deployment.Features.NginxFeature.Execute(RunningDeployment deployment)
12:34:34   Error    |       at Calamari.Deployment.Conventions.FeatureConventionBase.ExecuteFeatureClasses(RunningDeployment deployment, String feature)
12:34:34   Error    |       at Calamari.Deployment.Conventions.FeatureConventionBase.Run(RunningDeployment deployment)
12:34:34   Error    |       at Calamari.Deployment.ConventionProcessor.RunInstallConventions()
12:34:34   Error    |       at Calamari.Deployment.ConventionProcessor.RunConventions()
```

Reply:
This was due to the call to `GetConfigRootDirectory` being conditional on that there were additional locations. I've refactored the code to call `GetConfigRootDirectory` earlier in the `BuildConfiguration` method.